Makefile.detect

```diff
@@ -35,6 +35,9 @@ else
 ifeq ($(shell uname -o),GNU/Linux)
 PLATFORM=linux
 else
+ifeq ($(shell uname),SunOS)
+PLATFORM=solaris
+else
```

Comment:
Maybe keep the unix feature since solaris is a specific unix?

Reply:
And then check `uname` to guard `-lsocket`?  I'm not sure if it's appropriate on other things `unix` would apply to, since it's basically the default case.

Reply:
Sorry, not here, I meant in sexp.h at the Scheme level define both solaris and unix.

Reply:
Well, it's definitely better not to put "solaris" in there twice.  I've updated not to do that, at least.

I'm never sure what people think "unix" would apply to.  It seemed weird to me for "solaris" to imply "unix" but "bsd" not to, or "linux" not to, so I didn't do that bit.

Would
```c
#if !defined(_WIN32)
  "unix"
#endif
```
Be closer to the right thing? (until a VMS user comes along, I suppose...)

Reply:
Ah, nevermind, I thought unix was already defined there.  Looks like we set bsd along with the specific bsd but otherwise don't have multiple platform features.

The pr looks good.